src/ButtonGroup.php

```diff
@@ -4,143 +4,176 @@
 
 namespace Yiisoft\Yii\Bootstrap5;
 
-use Yiisoft\Arrays\ArrayHelper;
-use Yiisoft\Definitions\Exception\InvalidConfigException;
-use Yiisoft\Html\Html;
+use Yiisoft\Html\{Html, Tag\Div};
 
 use function implode;
-use function is_array;
+use function is_string;
 
 /**
  * ButtonGroup renders a button group bootstrap component.
  *
  * For example,
  *
  * ```php
- * // a button group with items configuration
  * echo ButtonGroup::widget()
- *     ->buttons([
- *         ['label' => 'A'],
- *         ['label' => 'B'],
- *         ['label' => 'C', 'visible' => false],
- *     ]);
- *
- * // button group with an item as a string
- * echo ButtonGroup::widget()
- *     ->buttons([
- *         Button::widget()->label('A'),
- *         ['label' => 'B'],
- *     ]);
+ *     ->addClass('btn-lg')
+ *     ->ariaLabel('Basic example')
+ *     ->buttons(
+ *         Button::widget()->label('Left')->id(false)->type(ButtonType::PRIMARY),
+ *         Button::widget()->label('Middle')->id(false)->type(ButtonType::PRIMARY),
+ *         Button::widget()->label('Right')->id(false)->type(ButtonType::PRIMARY),
+ *     )
+ *     ->render();
  * ```
  *
  * Pressing on the button should be handled via JavaScript. See the following for details:
  */
-final class ButtonGroup extends Widget
+final class ButtonGroup extends \Yiisoft\Widget\Widget
 {
+    private const NAME = 'btn-group';
+    private array $attributes = [];
+    /** psalm-var Button[] $buttons */
     private array $buttons = [];
-    private bool $encodeLabels = true;
-    private bool $encodeTags = false;
-    private array $options = [];
+    private string|null $cssClass = null;
+    private bool|string $id = true;
 
-    public function render(): string
+    /**
+     * Sets the CSS class attribute for the button group component.
+     *
+     * @param string $value The CSS class for the button group component (e.g., 'test-class').
+     *
+     * @return self A new instance of the current class with the specified class value.
+     *
+     * @link https://html.spec.whatwg.org/#classes
+     */
+    public function addClass(string $value): self
     {
-        if (!isset($this->options['id'])) {
-            $this->options['id'] = $this->getId();
-        }
+        $new = clone $this;
+        $new->cssClass = $value;
```

Comment:
Should it add a class or replace it?

Reply:
Added the possibility of adding the class without overwriting it.